File: Java/NetBeans/DSNV/src/dsnv/NVBC.java
package dsnv;

public class NVBC extends NV {
    protected float hsLuong;
    protected byte soNamCT;
    
    public NVBC(String hoTen, int phong, float hsLuong, byte soNamCT) {
        super(hoTen, phong);
        this.hsLuong = hsLuong;
        this.soNamCT = soNamCT;
    }
    
    public void hienThi() {
        super.hienThi();
        System.out.println("He so luong(trieu VND): " + hsLuong);
        System.out.println("So nam cong tac: " + soNamCT);
    }
    
    public String loaiNV() {
        return "BC";
    }
    
    public double layLuong() {
        return hsLuong * soNamCT;
    }
}

File: C#/DSNV/DSNV/NVBC.cs
namespace DSNV
{
    public class NVBC : NV
    {
        private float hsLuong;
        private int soNamCT;

        protected int SoNamCT
        {
            get { return soNamCT; }
            set { soNamCT = value; }
        }

        protected float HSLuong
        {
            get { return hsLuong; }
            set { hsLuong = value; }
        }

        public NVBC(string hoTen, int phong, float hsLuong, int soNamCT) : base(hoTen, phong)
        {
            this.hsLuong = hsLuong;
            this.soNamCT = soNamCT;
        }

        public override void hienThi()
        {
            base.hienThi();
            Console.WriteLine($"He so luong: {hsLuong}");
            Console.WriteLine($"So nam cong tac: {soNamCT}");
        }

        public override bool laNVHDDH()
        {
            return false;
        }

        public override float layLuong()
        {
            return soNamCT < 10 ? hsLuong * 2000000 : hsLuong * 4000000;
        }

        public override string loaiNV()
        {
            return "NVBC";
        }
    }
}
